component/discovery/transformer/transformer.go

```diff
@@ -0,0 +1,265 @@
+package transformer
+
+import (
+	"context"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/regexp"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/relabel"
+	"github.com/prometheus/prometheus/scrape"
+	"github.com/rfratto/gohcl"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "discovery.transformer",
+		Args:    Arguments{},
+		Exports: Exports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments holds values which are used to configure the discovery.transformer component.
+type Arguments struct {
+	// Targets contains the input 'targets' passed by a Service Discovery component.
+	Targets []Target `hcl:"targets,optional"`
+
+	// The following values can be used to modify the original target label set.
+	RelabelConfigs []*RelabelConfig `hcl:"relabel_configs,block"`
+	ScrapeLabels   ScrapeLabels     `hcl:"scrape_labels,optional"`
+	QueryParams    url.Values       `hcl:"query_params,optional"`
```

Comment:
I'm a little confused about what purpose these serve for transforming discovered targets. They both seem relevant to specifically metrics scraping and not just general discovery; should they be removed from here?